packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -1,82 +1,313 @@
-import { createRequire } from 'node:module'
 import { stringify } from 'csv-stringify/sync'
 import type {
     SuiteStats, Tag, HookStats, RunnerStats, TestStats, BeforeCommandArgs,
     AfterCommandArgs, CommandArgs, Argument
 } from '@wdio/reporter'
 import WDIOReporter from '@wdio/reporter'
 import type { Capabilities, Options } from '@wdio/types'
-
+import type { Label, MetadataMessage } from 'allure-js-commons'
+import { AllureRuntime, AllureGroup, AllureTest, AllureStep, Status as AllureStatus, Stage, LabelName, LinkType, ContentType, AllureCommandStepExecutable } from 'allure-js-commons'
 import {
-    addFeature, addLabel, addSeverity, addIssue, addTestId, addStory, addEnvironment,
-    addDescription, addAttachment, startStep, endStep, addStep, addArgument
+    addFeature, addLink, addOwner, addEpic, addSuite, addSubSuite, addParentSuite, addTag, addLabel, addSeverity, addIssue, addTestId, addStory, addEnvironment, addAllureId,
+    addDescription, addAttachment, startStep, endStep, addStep, addArgument, step,
 } from './common/api.js'
 import {
     getTestStatus, isEmpty, isMochaEachHooks, getErrorFromFailedTest,
-    isMochaAllHooks, getLinkByTemplate, attachConsoleLogs
+    isMochaAllHooks, getLinkByTemplate, findLast,
 } from './utils.js'
-import { events, PASSED, PENDING, SKIPPED, stepStatuses } from './constants.js'
+import { events } from './constants.js'
 import type {
     AddAttachmentEventArgs, AddDescriptionEventArgs, AddEnvironmentEventArgs,
     AddFeatureEventArgs, AddIssueEventArgs, AddLabelEventArgs, AddSeverityEventArgs,
-    AddStoryEventArgs, AddTestIdEventArgs, Status, AllureReporterOptions
+    AddEpicEventArgs, AddOwnerEventArgs, AddParentSuiteEventArgs, AddSubSuiteEventArgs,
+    AddLinkEventArgs, AddAllureIdEventArgs, AddSuiteEventArgs, AddTagEventArgs,
+    AddStoryEventArgs, AddTestIdEventArgs, AllureReporterOptions } from './types.js'
+import {
+    TYPE as DescriptionType
 } from './types.js'
 
-const require = createRequire(import.meta.url)
-
-/**
- * Allure v1 has no proper TS support
- * ToDo(Christian): update to Allure v2 (https://github.com/webdriverio/webdriverio/issues/6313)
- */
-const Allure = require('allure-js-commons')
-const Step = require('allure-js-commons/beans/step.js')
-
 export default class AllureReporter extends WDIOReporter {
-    private _allure: any
+    private _allure: AllureRuntime
     private _capabilities: Capabilities.RemoteCapability
     private _isMultiremote?: boolean
     private _config?: Options.Testrunner
-    private _lastScreenshot?: string
     private _options: AllureReporterOptions
     private _consoleOutput: string
     private _originalStdoutWrite: Function
-    private _addConsoleLogs: boolean
-    private _startedFeatures: SuiteStats[] = []
+
+    // TODO: i guess better to rename the property to make it easy to understand that we actually work with stack
```

Comment:
The name is okay. We don't always have to reflect the name with the behavior of the data structure. However, I think it would be better to move all the logic around this array (currentSuite, currentTest, currentAllureSpec, pop and push actions) to a separate class. It will reduce complexity of reporter class.

Reply:
:+1: